### PR TITLE
udptunnel: fix linkage to libnsl on Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
since this only impacts linux bottle, do we have some better way to just replacing that bottle instead of replacing the whole batch?

**Comment by ZhongRuoyu** [comment]
> since this only impacts linux bottle, do we have some better way to just replacing that bottle instead of replacing the whole batch?

Only seems to be possible if we manually delete the Linux bottle (both in the formula and on GitHub Packages), merge without bottles, and then rebuild for Linux only.

**Comment by chenrui333** [comment]
yeah, that is what I was a bit concerned about, but I guess we have to take the bite.

File: Formula/u/udptunnel.rb [hunk 1]
```diff
@@ -35,6 +35,8 @@ def install
     # Fix compile with newer Clang
     ENV.append_to_cflags "-Wno-implicit-function-declaration" if DevelopmentTools.clang_build_version >= 1200
 
+    ENV["LIBS"] = "-L/#{Formula["libnsl"].opt_lib}" if OS.linux?
```
Comment (carlocab) [inline Formula/u/udptunnel.rb:38]:
```suggestion
    ENV["LIBS"] = "-L#{Formula["libnsl"].opt_lib}" if OS.linux?
```

**Review by fxcoudert — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**